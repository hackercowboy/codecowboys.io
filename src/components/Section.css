.section {
  max-width: 750px;
  padding-bottom: 40px; }
  .section .section-title-wrapper {
    margin: 30px auto 10px auto;
    overflow: hidden;
    text-align: center; }
  .section h2, .section h3 {
    text-align: center; }
  .section h2:before, .section h2:after {
    content: "";
    vertical-align: top;
    display: inline-block;
    width: 50%;
    height: 0.65em;
    border-bottom: 1px solid #222222;
    margin: 2px 2% 0 -55%; }
  .section h2:after {
    margin: 2px -55% 0 2%; }
  .section h2 {
    font-size: 36px;
    padding-bottom: 0; }
  .section h3 {
    font-size: 26px;
    padding-bottom: 5px; }
  .section h4 {
    font-size: 20px;
    padding-top: 10px;
    padding-bottom: 15px; }

.section-dark {
  padding-top: 50px;
  margin-top: -50px;
  color: white;
  background: rgba(0, 0, 0, 0.6); }
  .section-dark h2:before, .section-dark h2:after {
    border-bottom: 1px solid white; }
