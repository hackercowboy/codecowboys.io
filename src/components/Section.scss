section {
  padding: 0 10px 40px;

  .section-title-wrapper {    
    margin: 30px auto 10px;
    overflow: hidden;
    text-align: center;
  }

  h2,
  h3 {
    text-align: center;
  }

  h2::before,
  h2::after {
    border-bottom: 1px solid #222;
    content: '';
    display: inline-block;
    height: 0.65em;
    margin: 2px 2% 0 -55%;
    vertical-align: top;
    width: 50%;
  }

  h2::after {
    margin: 2px -55% 0 2%;
  }

  h3 {
    padding-bottom: 5px;
    padding-top: 10px;
  }

  h4 {
    padding-bottom: 15px;
    padding-top: 10px;
  }
}

.section {
  max-width: 750px;
}

.section-light {
  background: #fff;
  padding-top: 5px;
}

.section-dark {  
  background: rgba(0, 0, 0, 0.6);
  color: #fff;
  margin-top: -50px;
  padding-top: 50px;

  h2::before,
  h2::after {
    border-bottom: 1px solid #fff;
  }
  
  a,
  a:visited {
    color: #fff;
  }
}
