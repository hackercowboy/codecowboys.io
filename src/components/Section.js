import React from 'react';
import PropTypes from 'prop-types';
import { Element } from 'react-scroll';
import Container from './Container';

import './Section.scss';

function Section({
  id, dark, title, subtitle, children,
}) {
  return (
    <section className={dark ? 'section-dark' : 'section-light'}>
      <Element
        name={id}
        duration={500}
        smooth="true"
      >
        <Container>
          <div className="section-title-wrapper">
            {title && (<h2>{title}</h2>)}
            {subtitle ? (<h3>{subtitle}</h3>) : null}
          </div>
          {children}
        </Container>
      </Element>
    </section>
  );
}

Section.propTypes = {
  children: PropTypes.node,
  dark: PropTypes.bool,
  id: PropTypes.string,
  subtitle: PropTypes.string,
  title: PropTypes.string,
};

Section.defaultProps = {
  children: undefined,
  dark: false,
  id: undefined,
  subtitle: undefined,
  title: undefined,
};

export default Section;
