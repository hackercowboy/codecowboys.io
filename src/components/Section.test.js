import React from 'react';
import { shallow } from 'enzyme';

import Section from './Section';

describe('<Section/>', () => {
  it('should render', () => {
    const wrapper = shallow(<Section />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
