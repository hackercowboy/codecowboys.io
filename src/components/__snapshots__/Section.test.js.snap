// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section/> should render 1`] = `
"<section id={[undefined]} className=\\"section-light\\">
  <Element name={[undefined]} duration={500} smooth=\\"true\\">
    <Container>
      <div className=\\"section-title-wrapper\\" />
    </Container>
  </Element>
</section>"
`;

exports[`<Section/> should render dark 1`] = `
"<section id={[undefined]} className=\\"section-dark\\">
  <Element name={[undefined]} duration={500} smooth=\\"true\\">
    <Container>
      <div className=\\"section-title-wrapper\\">
        <h2>
          test
        </h2>
        <h3>
          Test 2
        </h3>
      </div>
    </Container>
  </Element>
</section>"
`;
